packages/runtime/container-runtime/src/test/gc/gcConfigs.spec.ts

```diff
@@ -642,79 +565,102 @@ describe("Garbage Collection configurations", () => {
 		describe("shouldRunSweep", () => {
 			const testCases: {
 				shouldRunGC: boolean;
-				setSweepTimeout: boolean;
 				sweepEnabled: boolean;
-				runSweep?: boolean;
-				expectedResult: boolean;
+				shouldRunSweep?: boolean;
+				expectedShouldRunSweep: boolean;
 			}[] = [
 				{
 					shouldRunGC: false,
-					setSweepTimeout: true,
 					sweepEnabled: true,
-					runSweep: true,
-					expectedResult: false,
+					shouldRunSweep: true,
+					expectedShouldRunSweep: false,
 				},
 				{
 					shouldRunGC: true,
-					setSweepTimeout: false,
 					sweepEnabled: true,
-					runSweep: true,
-					expectedResult: false,
+					shouldRunSweep: true,
+					expectedShouldRunSweep: true,
 				},
 				{
 					shouldRunGC: true,
-					setSweepTimeout: true,
 					sweepEnabled: true,
-					runSweep: false,
-					expectedResult: false,
+					shouldRunSweep: false,
+					expectedShouldRunSweep: false,
 				},
 				{
 					shouldRunGC: true,
-					setSweepTimeout: true,
 					sweepEnabled: false,
-					runSweep: true,
-					expectedResult: true,
+					shouldRunSweep: true,
+					expectedShouldRunSweep: true,
 				},
 				{
 					shouldRunGC: true,
-					setSweepTimeout: true,
 					sweepEnabled: true,
-					expectedResult: true,
+					expectedShouldRunSweep: true,
 				},
 				{
 					shouldRunGC: true,
-					setSweepTimeout: true,
 					sweepEnabled: false,
-					expectedResult: false,
+					expectedShouldRunSweep: false,
 				},
 			];
-			testCases.forEach((testCase) => {
+			testCases.forEach((testCase, index) => {
 				it(`Test Case ${JSON.stringify(testCase)}`, () => {
 					injectedSettings[runGCKey] = testCase.shouldRunGC;
-					injectedSettings[runSweepKey] = testCase.runSweep;
-					injectedSettings[runSessionExpiryKey] = testCase.setSweepTimeout; // Sweep timeout is set iff sessionExpiry runs (under other default inputs)
+					injectedSettings[runSweepKey] = testCase.shouldRunSweep;
 					gc = createGcWithPrivateMembers(undefined /* metadata */, {
 						sweepAllowed: testCase.sweepEnabled,
 					});
 					assert.equal(
-						gc.shouldRunGC,
+						gc.configs.shouldRunGC,
 						testCase.shouldRunGC,
 						"PRECONDITION: shouldRunGC set incorrectly",
 					);
-					assert.equal(
-						gc.configs.sweepTimeoutMs !== undefined,
-						testCase.setSweepTimeout,
-						"PRECONDITION: sweep timeout set incorrectly",
-					);
 					assert.equal(
 						gc.configs.sweepEnabled,
 						testCase.sweepEnabled,
 						"PRECONDITION: sweepEnabled set incorrectly",
 					);
 					assert.equal(
 						gc.configs.shouldRunSweep,
-						testCase.expectedResult,
-						"shouldRunSweep not set as expected",
+						testCase.expectedShouldRunSweep,
+						`shouldRunSweep not set as expected`,
+					);
+				});
+			});
+		});
+		describe("sessionExpiryTimeoutMs and sweepTimeoutMs", () => {
```

Comment:
I feel like these cases are covered already but no harm in laying it out this way, it makes the point clearly.

Reply:
okay. Do you think we should leave a feature flag to disable session expiry? With this change, there is no way to do that except for disabling GC.
I think its fine but wanted to get your opinion.

Reply:
Added the setting back with default = session expiry enabled.